.github/workflows/report-integration-results.yml

```diff
@@ -0,0 +1,101 @@
+name: Report results of integration tests
+
+# using label for now, will think about this a little more
+on:
+  # temporary so I can test this
+  push:
+  pull_request:
+    types: [labeled]
+
+jobs:
+  report-results:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - id: get-github-access-token
+        uses: camertron/github-app-installation-auth-action@v1
+        with:
+          app-id: ${{ vars.PRIMER_ISSUE_TRIAGE_APP_ID_FOR_GITHUB }}
+          private-key: ${{ secrets.PRIMER_ISSUE_TRIAGE_APP_PRIVATE_KEY_FOR_GITHUB }}
+          client-id: ${{ vars.PRIMER_ISSUE_TRIAGE_APP_CLIENT_ID_FOR_GITHUB }}
+          client-secret: ${{ secrets.PRIMER_ISSUE_TRIAGE_APP_CLIENT_SECRET_FOR_GITHUB }}
+          installation-id: ${{ vars.PRIMER_ISSUE_TRIAGE_APP_INSTALLATION_ID_FOR_GITHUB }}
+      - id: check-if-branch-exists-in-gh
+        name: Check if branch exists in gh/gh
+        uses: actions/github-script@v7
+        env:
+          # BRANCH: "prc-integration-test-for-${{github.event.pull_request.number}}"
+          BRANCH: "prc-integration-test-for-4358-failing-case"
+        with:
+          # github-token: ${{ steps.get-github-access-token.outputs.access-token }}
+          github-token: ${{ secrets.TEMP_SID_PAT }}
+          script: |
+            const { BRANCH } = process.env;
+            try {
+              const { data } = await github.rest.repos.getBranch({
+                owner: 'github',
+                repo: 'github',
+                branch: BRANCH
+              });
+              core.setOutput('branch', BRANCH);
```

Comment:
Instead of us calling out to the repo directly, do you think it would be possible for us to use repository dispatch to coordinate between github/github and primer/react? The advantage of this would be that we wouldn't need to have a token/app that has permissions to github/github in CI.

Reply:
Definitely possible! 👀 Hopefully we can make them as granular. 

> The advantage of this would be that we wouldn't need to have a token/app that has permissions to github/github in CI.

We would still need a token that can do the first repository dispatch to trigger `primer-react-pr-test` workflow (not included in this PR). But, that would need less permissions which is definitely better

Reply:
> The advantage of this would be that we wouldn't need to have a token/app that has permissions to github/github in CI.

How worried should we be about this?